substrate/frame/transaction-storage/src/lib.rs

```diff
@@ -432,8 +444,8 @@ pub mod pallet {
 			let byte_fee = ByteFee::<T>::get().ok_or(Error::<T>::NotConfigured)?;
 			let entry_fee = EntryFee::<T>::get().ok_or(Error::<T>::NotConfigured)?;
 			let fee = byte_fee.saturating_mul(size.into()).saturating_add(entry_fee);
-			ensure!(T::Currency::can_slash(&sender, fee), Error::<T>::InsufficientFunds);
-			let (credit, _) = T::Currency::slash(&sender, fee);
+			T::Currency::hold(&HoldReason::StorageFeeHold.into(), &sender, fee)?;
+			let (credit, _) = T::Currency::slash(&HoldReason::StorageFeeHold.into(), &sender, fee);
```

Comment:
Seems logically equivalent to just reducing the account balance by `fee` if the balance is gt `fee`? Or am I missing something?

Reply:
You could very well use `decrease_balance` which returns the reduced balance. I don't see why that wouldn't work.
Though, I don't know why it was not used in the original code